### PR TITLE
Propagate python exceptions through Wasm

### PR DESCRIPTION
**Pull request by Amanieu**
Depends on https://github.com/wasmerio/wasmer/pull/2728

### CODE REVIEW

File: packages/api/src/externals/function.rs [hunk 1]
```diff
@@ -199,9 +199,7 @@ impl Function {
                 let results = environment
                     .py_function
                     .call(py, PyTuple::new(py, arguments), None)
-                    .map_err(|error| {
-                        wasmer::RuntimeError::new(io::Error::from(error).to_string())
-                    })?;
+                    .map_err(|error| wasmer::RuntimeError::custom(Box::new(error)))?;
```
Comment (syrusakbary) [inline packages/api/src/externals/function.rs:202]: Probably not need to map it any longer?

Comment (Amanieu) [reply packages/api/src/externals/function.rs:202]: This is necessary to pass the `PyErr` through the `RuntimeError` in wasmer. The closure passed to `Function::new_with_env` must have an error type of `RuntimeError`.

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**